test/browser-plugins/plugins.test.js

```diff
@@ -181,79 +186,185 @@ describe('Plugins', () => {
+
+                    await browser.close();
+
+                    const launchContext2 = plugin.createLaunchContext({ useIncognitoPages: true });
+                    browser = await plugin.launch(launchContext2);
+                    expect(browser).not.toBeInstanceOf(PlaywrightBrowser);
+                });
+
+                test('should return correct version', async () => {
+                    const plugin = new PlaywrightPlugin(playwright[browserName]);
+
+                    const launchContext = plugin.createLaunchContext({ useIncognitoPages: false });
+                    browser = await plugin.launch(launchContext);
+                    const version1 = browser.version();
+
+                    await browser.close();
+
+                    const launchContext2 = plugin.createLaunchContext({ useIncognitoPages: true });
+                    browser = await plugin.launch(launchContext2);
+                    expect(version1).toEqual(browser.version());
+                });
+
+                test('should return all contexts', async () => {
+                    const plugin = new PlaywrightPlugin(playwright[browserName]);
+
+                    const launchContext = plugin.createLaunchContext();
+                    browser = await plugin.launch(launchContext);
+                    const contexts = browser.contexts();
+                    expect(contexts).toHaveLength(1);
+                    expect(contexts[0]).toEqual(browser.browserContext);
+                });
+
+                test('should return correct connected status', async () => {
+                    const plugin = new PlaywrightPlugin(playwright[browserName]);
+
+                    const launchContext = plugin.createLaunchContext();
+                    browser = await plugin.launch(launchContext);
+                    expect(browser.isConnected()).toBeTruthy();
+
+                    await browser.close();
+
+                    expect(browser.isConnected()).toBeFalsy();
+                });
+
+                test('should return old context as new context and log error', async () => {
+                    const plugin = new PlaywrightPlugin(playwright[browserName]);
+                    jest.spyOn(log, 'error');
+                    const launchContext = plugin.createLaunchContext();
+                    browser = await plugin.launch(launchContext);
+                    const context = await browser.newContext();
+
+                    expect(context).toEqual(browser.contexts()[0]);
+                    expect(log.error).toBeCalledWith('Could not call `newContext()` on browser, when `useIncognitoPages` is set to `false`');
+                });
+
+                test('should have same public interface as playwright browserType', async () => {
+                    const plugin = new PlaywrightPlugin(playwright[browserName]);
+                    const originalFunctionNames = ['close', 'contexts', 'isConnected', 'newContext', 'newPage', 'version'];
```

Comment:
Would be great if we could make this generic. It would pull the functions from the real Playwright browser and we would test against those. It would help us catch changes automatically.

Reply:
I have originally tried this, but there are more private methods without the conventional `_`. I couldn't find any generic solution to do this 😞.

Reply:
Shame 😞. Let's check it again when we need to rewrite this 😄